### PR TITLE
The PR

### PR DESCRIPTION
**Pull request by hacdias**
Track on https://github.com/ipfs-shipyard/ipfs-desktop/issues/669.

### CODE REVIEW

**Comment by hacdias** [comment]
@olizilla I've just pushed the changes I told you about on IRC today: settings are finally in a window other than the menubar (it opens really fast too 😄).

Some background that might be useful about the contents of [`src`](https://github.com/ipfs-shipyard/ipfs-desktop/tree/new/src) directory:

- `hooks` are mainly stuff that run when something sends an event to ipcMain.
- `icons` are icons are icons.
- `utils` just some utils.
- `windows` is where everything gets interesting:
	- there's a `common` folder with common bundles, components and other files that might be common.
	- then there's a folder and a `.js` file with the same name for each different window. Inside the directory you can find the React app that will be run once the window opens. The `.js` file creates the window and takes care of some related events.

Hope this was a good brief explanation of the code. Not sure if this is the best code/structure design to do this, but it seemed right. What do you think?

**Comment by hacdias** [comment]
image

@ipfs-shipyard/gui getting more similar to @olizilla's suggestion:https://github.com/ipfs/ipfs-gui/issues/44#issuecomment-432998304

**Comment by hacdias** [comment]
@olizilla I feel we should merge this ASAP. This is getting too big and hard to manage. As soon as it is on master, it will be easier to work on.

**Comment by hacdias** [comment]
Try now @lidel

**Comment by diasdavid** [comment]
OMG WOWZA, when is this coming out? :D

**Comment by hacdias** [comment]
@diasdavid soon™

**Comment by lidel** [comment]
@hacdias  


### Daemon starts fine, but Desktop fails to open Web UI 

Tray icon appears, when I click on it [popup window looks fine as well](https://user-images.githubusercontent.com/157609/48344480-75fee000-e675-11e8-8384-acbb10572a64.png), however when I open webui screens I get white window and its console (<kbd>Ctrl+Shift+I</kbd>) contains [this error](https://user-images.githubusercontent.com/157609/48344547-b4949a80-e675-11e8-8ae3-518546837481.png).

I noticed that electron version on my system was `electron-1.6.18`, so perhaps that is the reason?
To which version should I update? 
Perhaps we should enforce minimal version and fail to start if it is too old? 

### Other UX issues

Apart from the above blocker, this PR has some minor UX regressions if user had some prior/custom setup, but that can be fixed in separate PRs.  To make it easier to track I created dedicated issues:

- Hangs on repo update (missing automatic migration):  https://github.com/ipfs-shipyard/ipfs-desktop/issues/687
- Missing support for external daemon already running:  https://github.com/ipfs-shipyard/ipfs-desktop/issues/688
- Fails to start when ~/.ipfs/api exists:  https://github.com/ipfs-shipyard/ipfs-desktop/issues/689

**Comment by fsdiogo** [comment]
I'm having the same issue as @lidel, I'm able to run the app but when opening Web UI that error is occurring.

**Comment by olizilla** [comment]
@lidel @fsdiogo have you run a `npm run build` ? you need to get your local copy to fetch a webui. It should be documented, and possibly checked as part of the start script to simplify the onboarding, but that should fix it for you for now.

**Comment by lidel** [comment]
I did run `build`, but `src/windows/webui/app/` was already in place AND empty because preload nodes not longer support get and failed silently. I fixed that in https://github.com/ipfs-shipyard/ipfs-desktop/pull/662/commits/9e425ea09efd317b425bd9da613b778d0745df24 but still no success, [same error](https://user-images.githubusercontent.com/157609/48371800-0b719280-e6bd-11e8-94aa-b5c0b3f4038e.png).


@hacdias question: why do we re-compile ipfs-webui using babel?  AFAIK it is already minified and chunked. 

https://github.com/ipfs-shipyard/ipfs-desktop/blob/9e425ea09efd317b425bd9da613b778d0745df24/package.json#L17-L21

**Comment by lidel** [comment]
@hacdias that fixed the "white screen" issue I had, thanks! Web UI looks good now. :+1: 

While I was looking at that, noticed something else:

###  ipfs-css icons are in git :thinking: 

I see we are using icons from `ipfs-css` but we have a build step `build:icons` that converts them to `.js` files and those >100 files are committed to the repo, artificially  ballooning the changeset. 

We run that conversion step every time `npm run build` is executed, so I don't think we should  be tracking those `.js` files. It is enough to have `ipfs-css` version in `package.json`. 

I think before this PR is merged:
1. Create a commit that removes generated `src/icons/*.js` from git
2. Update `build:icons` step to use different directory for generated icons, eg. `src/icons/ipfs-css` and add that directory to `.gitignore` so they won't be tracked by git. 

Does that sounds good, or am I missing something?

**Comment by olizilla** [comment]
This is copied from web UI. I'm not super fussed about the number of files in the repo if it improves the device experience ("import the icon and you're done") and files that nothing imports don't get included in the bundler (that is important to me)

That said, if you feel strongly, I could be pursuaded that it's not much effort to convert the svgs only on a case by case basis, so we end up with just the set we are using currently. This adds a little device friction to adding a new one, but not much.

Also of note, create-react-app v2 was looking at supporting this sort of conversion in we pack config so it may be something to review more generally.

**Comment by olizilla** [comment]
The vertical alignment, wrapping, and spacing in the menu items needs some love...
screenshot 2018-11-14 at 10 32 06

File: src/hooks/ipfs.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    menubarWindow.it.tray.setImage(logo('black'))
+    menubarWindow.it.tray.setToolTip('IFPS stopped')
+  }
+
+  conn.running ? onStarted() : onStopped()
+  conn.on('started', onStarted)
+  conn.on('stopped', onStopped)
+
+  setInterval(getPeers(opts), 5000)
+  setInterval(getRepoSize(opts), 5000)
```
[the remark this answers is not in the document]
Comment (diasdavid) [reply src/hooks/ipfs.js]: Perhaps have a button for `refresh`

[the remark this answers is not in the document]
Comment (hacdias) [reply src/hooks/ipfs.js]: I reduced the time for now, but we could think about a refresh button!

**Pull request merged**